index.html

```diff
@@ -1690,59 +1690,93 @@ <h3>Generate Proof</h3>
       </section>
 
       <section>
-        <h3>Verify Proof Algorithm</h3>
-        <p class="issue">
-This algorithm is highly specific to digital signatures and needs to be
-generalized to other proof mechanisms such as Equihash.
-        </p>
+        <h3>Verify Proof</h3>
 
         <p>
 The following algorithm specifies how to check the authenticity and
-integrity of a <a>signed data document</a> by verifying its
-digital proof. This algorithm takes a
-<a>signed data document</a>, <var>signed document</var> and
-outputs a <code>true</code> or <code>false</code> value based on whether or
-not the digital proof on <var>signed document</var> was verified. Whenever
-this algorithm encodes strings, it MUST use UTF-8 encoding.
+integrity of a <a>secured data document</a> by verifying its
+digital proof. Required inputs are a
+<a>secured data document</a> (<var>securedDocument</var>) and
+<a>proof options</a> (<var>options</var>). A <dfn>verification result</dfn> is
+produced as output.
         </p>
 
-<div class="issue">Specify how the public key can be obtained, through
-some out-of-band process and passed in or it can be retrieved by
-derefencing its URL identifier, etc.</div>
-
         <ol class="algorithm">
           <li>
-Get the <a>public key</a> by dereferencing its URL identifier in the
-<code>proof</code> node of the default graph of <var>signed document</var>.
-Confirm that the <a>unsigned data document</a> that describes the <a>public
-key</a> specifies its controller and that its controllers's URL identifier can
-be dereferenced to reveal a bi-directional link back to the key. Ensure that the
-key's controller is a trusted entity before proceeding to the next step.
+Let <var>proof</var> be set to <var>securedDocument</var>.<var>proof</var>.
+          </li>
+          <li>
+If the <var>proof</var>.<var>type</var>,
+<var>proof</var>.<var>verificationMethod</var>, or
+<var>proof</var>.<var>proofPurpose</var> values are not set, a
+`MALFORMED_PROOF_ERROR` MUST be raised.
+          </li>
+          <li>
+If the cryptographic suite does not support unlinkability and the
```

Comment:
Maybe we just want to say "if the cryptographic suite requires `proof.created`" rather than make it hinge on the unlinkability property.